tensorflow/python/training/checkpointable/util_test.py

```diff
@@ -1324,6 +1324,21 @@ def test_initialize_if_not_restoring(self):
       train_fn()
       self.assertEqual(42., self.evaluate(optimizer.variables()[0]))
 
+  @test_util.run_in_graph_and_eager_modes
+  def test_restore_after_adding_empty_checkpointable_data_structure(self):
+    with ops.Graph().as_default(), self.session(graph=ops.get_default_graph()):
```

Comment:
Maybe I misunderstand what you are saying, but doesn't it require a new graph for a clean restore and a new session as well?

Reply:
How do I reset the graph when running in eager?

Reply:
Like so?
```python
  @test_util.run_in_graph_and_eager_modes
  def test_restore_after_adding_empty_checkpointable_data_structure(self):
    model = NonLayerCheckpointable()
    checkpoint = checkpointable_utils.Checkpoint(model=model)
    checkpoint.restore(None).initialize_or_restore()
    checkpoint_directory = self.get_temp_dir()
    checkpoint_prefix = os.path.join(checkpoint_directory, "ckpt")
    save_path = checkpoint.save(checkpoint_prefix)
    ops.reset_default_graph()
    model = NonLayerCheckpointable()
    model.dict = {"a": 1}
    model.list = {"b": 1}
    load_status = checkpoint.restore(save_path)
    load_status.assert_existing_objects_matched().run_restore_ops()
```

Reply:
There isn't really a Graph to reset. I'd just take out the lines; no need to reset. You can `del model` if you want.

And I see a few existing tests in this file are accidentally not running executing eagerly for the same reason, probably because they predate the as_default() -> graph mode change; I'll follow up on that in a different change.

Reply:
Looking at `@test_util.run_in_graph_and_eager_modes` this should work. I have to leave for now. If anything still comes up I can fix it later today.